Source/Dafny/Translator.cs

```diff
@@ -10321,7 +10546,16 @@ void TrStmt(Statement stmt, BoogieStmtListBuilder builder, List<Variable> locals
           if (assertStmt != null && assertStmt.Proof != null) {
             proofBuilder = new BoogieStmtListBuilder(this);
             AddComment(proofBuilder, stmt, "assert statement proof");
-            TrStmt(((AssertStmt)stmt).Proof, proofBuilder, locals, etran);
+            if (assertStmt.Grobner) {
+              AddComment(proofBuilder, stmt, "assert by Grobner");
+              var singularDriver = new SingularDriver(stmt);
+              // append assume false, the toks are messed up?
+              var falseBody = new LiteralExpr(stmt.Tok, false);
+              falseBody.Type = new BoolType();
+              var assumeFalse = new AssumeStmt(stmt.Tok, stmt.EndTok, falseBody, null);
+              assertStmt.Proof.Body.Add(assumeFalse);
+            }
+            TrStmt(assertStmt.Proof, proofBuilder, locals, etran);
```

Comment:
Just to confirm, this is a Dafny level change that adds a Dafny-level "assume false" to the proof body that's supplied in the `by { ... }` clause of the assert?  And it's added at the end of the body, so that Dafny will check the assertions the developer wrote, since Singular trusts that they are correct?